src/plugins/intel_gpu/tests/unit/dynamic_execution/skip_gather_at_runtime.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "test_utils.h"
+
+#include <intel_gpu/primitives/input_layout.hpp>
+#include <intel_gpu/primitives/gather.hpp>
+
+#include "program_wrapper.h"
+
+#include <cmath>
+#include <algorithm>
+
+using namespace cldnn;
+using namespace ::tests;
+
+namespace skip_gather_tests {
+TEST(skip_gather_at_runtime, not_skip_if_cpuimpl) {
+    auto& engine = get_test_engine();
+
+    auto in1_layout = layout{ov::PartialShape{-1, 32, -1, 128}, data_types::f32, format::bfyx};
+    auto in2_layout = layout{ov::PartialShape{-1}, data_types::i32, format::bfyx};
+
+    topology topology(input_layout("input1", in1_layout),
+                      input_layout("input2", in2_layout),
+                      gather("gather",
+                             input_info("input1"),
+                             input_info("input2"),
+                             0,                                       // axis
+                             in1_layout.get_partial_shape().size(),   // input rank
+                             ov::Shape{},                             // output shape
+                             0,                                       // batch_dim
+                             true),                                   // support_neg_ind
+                      reorder("reorder", input_info("gather"), format::bfyx, data_types::f32)); /*output padding*/
+
+    auto config = get_test_default_config(engine);
+    config.set_property(ov::intel_gpu::allow_new_shape_infer(true));
+    config.set_property(ov::intel_gpu::force_implementations(ov::intel_gpu::ImplForcingMap{ {"gather", {format::bfyx, "", impl_types::cpu}} }));
+
+    network network(engine, topology, config);
+    auto gather_inst = network.get_primitive("gather");
+    ASSERT_EQ(gather_inst->can_be_optimized(), false);
+}
+}  // skip_gather_tests
```

Comment:
This is weird because the test is "skip_gather_at_runtime" but there is no "skipping" case....
1. Please add another test that actually skips
2. Also pleas add a check that actually executes the network and compares outputs
 (please refer to dynamic_execution/stateful_model.cpp::skip_gather_at_runtime) 

**Or**, I think you can just add another test in dynamic_execution/stateful_model.cpp, which has cpu gather, instead of adding this new file.

Reply:
Modified skip_gather_at_runtime to not_skip_gather_in_cpuimpl. Thanks for your comments.